tests/GeneratedHydratorTest/CodeGenerator/Visitor/HydratorMethodsVisitorTest.php

```diff
@@ -81,4 +83,106 @@ public function classAstProvider() : array
             [$staticClassName, $parser->parse('<?php ' . $staticClassCode)[0], ['taz']],
         ];
     }
+
+	/**
+	 * @dataProvider propertyAstProvider
+	 */
+	public function testPropertyCodeGeneration(string $className, Class_ $classNode, array $properties) : void
+	{
+		$visitor = new HydratorMethodsVisitor(new ReflectionClass($className));
+
+		/** @var Class_ $modifiedAst */
+		$modifiedNode = $visitor->leaveNode($classNode);
+
+		$constructors = $this->findConstructor($modifiedNode);
+		self::assertCount(1, $constructors);
+		$constructor = reset($constructors);
+
+		$hydratePropertyNames = $this->findAssignedPropertyNames(
+			$constructor,
+			'hydrateCallbacks',
+			'object',
+			function (Assign $assign) {
+				return $assign->var->name->name;
+			}
+		);
+		$extractPropertyNames = $this->findAssignedPropertyNames(
+			$constructor,
+			'extractCallbacks',
+			'values',
+			function (Assign $assign) {
+				return $assign->var->dim->value;
+			}
```

Comment:
Not important, but `\t` is generally to be avoided. Can fix on merge from my end though.

Reply:
ah sorry. didn't switch to spaces (working with tabs internally).

Reply:
No worries: that is entirely my preference, and I just forgot to add PHPCS to the repo ;-)